src/main/webapp/app/shared/util/firebase/firebase-history-utils.ts

```diff
@@ -301,7 +305,7 @@ export function makeFirebaseKeysReadable(keys: string[]) {
 export function makeHistoryLocationReadable(locationParts: string[], drugList: readonly IDrug[]) {
   const readableKeys: string[] = [];
   for (const part of locationParts) {
-    if (Object.values(TI_TYPE_TO_HISTORY_STRING).includes(part)) {
+    if ([...Object.values(TI_TYPE_TO_HISTORY_STRING), ...Object.values(TI_TYPE_TO_LEGACY_HISTORY_STRING)].includes(part)) {
```

Comment:
Is this called a lot? It's probably small performance hit, but could you make this array a constant outside the function?

Reply:
Yeah for sure, good catch